### PR TITLE
Ignores some cppcheck warnings

### PR DESCRIPTION
**Pull request by Xav83**
This Pull Request adds comments made so that, when using `inline-suppr` option, cppcheck will ignore the warnings about not using explicit keyword when a constructor has one parameter.

It addresses some of the warnings in #1759 

* * *

## Pull request checklist

Read the [Contribution Guidelines](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md) for detailed information.

- [ ]  Changes are described in the pull request, or an [existing issue is referenced](https://github.com/nlohmann/json/issues).
- [ ]  The test suite [compiles and runs](https://github.com/nlohmann/json/blob/develop/README.md#execute-unit-tests) without error.
- [ ]  [Code coverage](https://coveralls.io/github/nlohmann/json) is 100%. Test cases can be added by editing the [test suite](https://github.com/nlohmann/json/tree/develop/test/src).
- [ ]  The source code is amalgamated; that is, after making changes to the sources in the `include/nlohmann` directory, run `make amalgamate` to create the single-header file `single_include/nlohmann/json.hpp`. The whole process is described [here](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md#files-to-change).

## Please don't

- The C++11 support varies between different **compilers** and versions. Please note the [list of supported compilers](https://github.com/nlohmann/json/blob/master/README.md#supported-compilers). Some compilers like GCC 4.7 (and earlier), Clang 3.3 (and earlier), or Microsoft Visual Studio 13.0 and earlier are known not to work due to missing or incomplete C++11 support. Please refrain from proposing changes that work around these compiler's limitations with `#ifdef`s or other means.
- Specifically, I am aware of compilation problems with **Microsoft Visual Studio** (there even is an [issue label](https://github.com/nlohmann/json/issues?utf8=✓&q=label%3A%22visual+studio%22+) for these kind of bugs). I understand that even in 2016, complete C++11 support isn't there yet. But please also understand that I do not want to drop features or uglify the code just to make Microsoft's sub-standard compiler happy. The past has shown that there are ways to express the functionality such that the code compiles with the most recent MSVC - unfortunately, this is not the main objective of the project.
- Please refrain from proposing changes that would **break [JSON](http://json.org) conformance**. If you propose a conformant extension of JSON to be supported by the library, please motivate this extension.
- Please do not open pull requests that address **multiple issues**.

### CODE REVIEW

**Comment by Xav83** [comment]
Hi @nlohmann 😃 

First of all, thanks for your interest in this Pull Request. 👍 

I agree with you that this is not something really elegant.
But if someone uses cppcheck on a project including the json library, he should not have those warnings too (which is my personal objective, with the issue #1759).

There are other ways to suppress warnings in cppcheck, detailed in the [manual](http://cppcheck.sourceforge.net/manual.pdf).

We can suppress each warnings in the command line and given this line to the cppcheck users.
But the more warnings will be ignored, the less this command line will be easy to read, or to integrate in a project using the json library and cppcheck.

We can create a text file, or a xml file in which the ignored warning will be detailed and given to cppchek, so that it won't trigger them. If this solution is used, then, it should be well described in the README, to make it easy for the cppcheck users to integrate this file in their workflow(since they can integrate it from conan, or as a cmake external project, for example).

I would happily modify this PR to use another solution. 😃 
But I will need you, @nlohmann , or someone else to validate which solution to integrate.

Thank you in advance 🙂

**Comment by trraney328** [comment]
@Xav83 I went ahead and reviewed it

File: include/nlohmann/detail/input/input_adapters.hpp [hunk 1]
```diff
@@ -374,6 +380,7 @@ class input_adapter
                  std::is_integral<typename std::remove_pointer<CharT>::type>::value and
                  sizeof(typename std::remove_pointer<CharT>::type) == 1,
                  int>::type = 0>
+    // cppcheck-suppress noExplicitConstructor
     input_adapter(CharT b)
         : input_adapter(reinterpret_cast<const char*>(b),
                         std::strlen(reinterpret_cast<const char*>(b))) {}
```
Comment (trraney328) [inline include/nlohmann/detail/input/input_adapters.hpp:386]: this is the reason why

**Review by trraney328 — COMMENTED** [review]
you may have to fix line 386

**Pull request closed without merging**